api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -751,21 +753,34 @@ public User getUserByActivationKey(String activationKey) {
 	}
 	
 	/**
+	 * @throws MessageException
 	 * @see org.openmrs.api.UserService#setUserActivationKey(org.openmrs.User)
-	 * 
 	 */
 	@Override
 	public User setUserActivationKey(User user) {
 		String token = RandomStringUtils.randomAlphanumeric(20);
 		long time = System.currentTimeMillis() + getValidTime();
 		String hashedKey = Security.encodeString(token);
-		String activationKey = hashedKey+":"+time;
+		String activationKey = hashedKey + ":" + time;
 		LoginCredential credentials = dao.getLoginCredential(user);
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
-			
-		//Send Email with unhashed  activation key and Date:
 		
+		String link = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_HOST_URL)
+		        .replace("activationkey", "activationkey=" + token);
+		String msg = "Hi " + user.getUsername()
+		        + ",\n It looks like you want to change your password, click on the following link to do so.\n"
+		        + link
+		        + "\n Please disregard this email if you didn't innitiate the password reset process.  \n The openmrs Team";
+		
+		try {
+			Message mail = Context.getMessageService().createMessage(user.getEmail(),
+			    Context.getAdministrationService().getGlobalProperty("mail.from"), "Password Reset Request", msg);
+			Context.getMessageService().sendMessage(mail);
```

Comment:
There is actually a sendMessage method that takes in recipients, from, subject and content, I think you need to call that.